### ISSUE TITLE
Support web sockets

### ISSUE THREAD

**Issue by kolektiv** [freya-fs/freya#93]
Come up with a decent design/library for supporting web sockets. Not totally sure how they should be mounted/maintained in terms of routing - they're not really something that fits as a pipeline? But yeah, would be good to have pretty soon in some form.

**Comment by panesofglass** [comment]
I think we just have to model the protocol upgrade from HTTP.

**Comment by panesofglass** [comment]
What do you think, 1.0 or 1.1? Perhaps 1.0 for handling the upgrade? If so, I think we would also want to do something about HTTP Streaming and Server-sent events. I found [this resource](http://enterprisewebbook.com/ch8_websockets.html) helpful. My current vote is push this to 1.*.

**Comment by kolektiv** [comment]
Yeah 1.1 is probably sane, I'd probably be hopeful of calling what's there
now 1.0 I think, and new features go in to further releases. I think it's
pretty close to sane, and there's no reason we can't stick to semver and
develop from a 1.0 soon.

**Comment by panesofglass** [comment]
Sure! Let's nail down some docs and present it to the world!

**Comment by panesofglass** [comment]
This needs two parts: the pipeline piece and the machine extension.
